File: MergeTwoSortedLinkedList.java
class Solution {
    public static void main(String[] args) {
        ListNode list1 = Utility.createSampleListNode();
        ListNode list2 = Utility.createSampleListNode();
        Utility.printListNode(mergeTwoSortedListNode(list1, list2));
    }

    public static ListNode mergeTwoSortedListNode(ListNode list1, ListNode list2) {
        ListNode dummyListNode = new ListNode(-1);
        ListNode curr = dummyListNode;
        while (list1 != null && list2 != null) {
            if (list1.val <= list2.val) {
                curr.next = list1;
                list1 = list1.next;
            } else {
                curr.next = list2;
                list2 = list2.next;
            }
            curr = curr.next;
        }

        if (list1 != null) {
            curr.next = list1;
        } else if (list2 != null) {
            curr.next = list2;
        }

        return dummyListNode.next;
    }
}

File: merge_two_sorted_linked_list.py
from typing import Optional
from list_node import ListNode
from list_node_utils import ListNodeUtils


class Solution:
    def mergeTwoLists(self, list1: Optional[ListNode], list2: Optional[ListNode]) -> Optional[ListNode]:
        dummy = ListNode(-1)
        cur = dummy
        while list1 and list2:
            if list1.val <= list2.val:
                cur.next = list1
                list1 = list1.next
            else:
                cur.next = list2
                list2 = list2.next
            cur = cur.next
            
        if list1:
            cur.next = list1
        elif list2:
            cur.next = list2
        
        return dummy.next
    
    
list1 = ListNodeUtils.create_linked_list([1, 2, 4])
list1.print()
list2 = ListNodeUtils.create_linked_list([1, 3, 5])
list2.print()
solution = Solution()
merge_list = solution.mergeTwoLists(list1, list2)
merge_list.print()